### PR TITLE
DNM splitting into separate PRs Review ready: various enhancements to the deployment

### PR DESCRIPTION
**Pull request by relyt0925**
Summary below (very open to refactoring/doing it the way y'all prefer just wanted to get something working first):

1) Provide ability for different service publishing strategies. The default approach uses 2 load balancers per cluster. This also adds a model where the infrastructure services can be exposed as a node port and if a centralized load balancer exposes the entire node port range  exists traffic can flow through that. It means that the number of total load balancers in use does not scale with the number of clusters (it remains constant: one load balancer per zone exposing the node port range)

2) Instead of hard coding what the APIServerSecurePort is and the AdvertisedIPAddress we now allow that to be specified through the HostedClusterCRD.

3) Allow the ability to enable and disable asset payloads. Was thinking this will be useful depending on the environment/provider a customer is targeting (AWS/IBMCloud/Azure etc). Can also help for toggling a VPN deployment versus a konnectivity style deployment

4) Integrates a haproxy side car server with the machine config server that can enable it to do authentication using the node bootstrapper token. 

5) Adds a controller that handles syncing any changes to the node bootstrapper token in a user cluster to the control plane and updating of the asssociated user-data secret to update the auth for the initial call workers use to auth to the ignition server.

6) Moves the ignition route to have E2E tls validation. CAs are passed to the user-data that workers use to make the initial ignition call to validate TLS.

Then just one non important question that we can talk about the next time we sync:

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go [hunk 1]
```diff
@@ -393,23 +397,57 @@ func (r *HostedControlPlaneReconciler) ensureInfrastructure(ctx context.Context,
...
+	switch hcp.Spec.ControlPlaneServiceTypeStrategy {
+	case "NodePort":
+		status.APIAddress = hcp.Spec.ControlPlaneNodePortIngressTrafficDomain
+		status.APIPort = apiAddress
+	default:
+		status.APIAddress = apiAddress
+		status.APIPort = fmt.Sprint(APIServerPort)
+	}
+
+	vpnEnabled := true
```
Comment (relyt0925) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:419]: this seems hacky to me but thought I would try it for a protoype to show what I mean by "payload enablement/disablement"

**Comment by sjenning** [comment]
Good work @relyt0925 !

I would love to see this broken down into different PRs.  I've personally been wanting 1 for a while now and can be a self-contained changed.  4-6 look like they could be bunched together in a PR as they are all part of "authenticating the MCS".  2 and 3 can be their own PRs as well.

So could you break this down into 4 PRs?  1, 2, 3, and 4-6?

> Then just one non important question that we can talk about the next time we sync:

What is the question?

**Comment by relyt0925** [comment]
Hey @sjenning! Will do 

The question is:
https://github.com/openshift/hypershift/pull/86/files#diff-3d4ffda0834e09a59a4d510523932e20b9234bcb9816e40258a0fb32a87a7006R1068

```
//TL: I believe this we will want to change so we don't get in a case of subdomains being hijacked from the control plane in user clusters.
//the "management cluster" and the clusters it hosts should be independent from a DNS perspective. For example: one management
//cluster can host clusters from a variety of different tenants and sometimes different providers (IBM Cloud VPC Gen 2 and IBM Cloud Classic in same tugboat)
// We likely want a user to pass this in explicitly and some automated system can use the data they pass in and adjust the base domain to point to the proper
// router pod ips when applicable.
```

**Comment by relyt0925** [comment]
First PR separated is here:
https://github.com/openshift/hypershift/pull/89

**Comment by relyt0925** [comment]
part 2 is here:
https://github.com/openshift/hypershift/pull/91

**Comment by relyt0925** [comment]
this is part 3:
https://github.com/openshift/hypershift/pull/92

**Comment by sjenning** [comment]
This PR has been decomposed into other PRs/Issues.  It is overly board and now requires a huge rebase.

**Pull request closed without merging**